### PR TITLE
Fix several issues related with disaggregate remote read

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/6827

Close https://github.com/pingcap/tiflash/issues/7052 (but I have no idea why it is fixed)

Problem Summary:

### What is changed and how it works?

Please review commit by commit.

After this PR, most of the issues related with disaggregated read have been resolved, except for this one, which occurs sometimes:

```
execute run failed, err execute query q1 failed Error 1105: other error for mpp stream: From MPP<query:<query_ts:1678662934459371000, local_query_id:302, server_id:3258449, start_ts:440051416283742219>,task_id:1>: Poco::Exception. Code: 1000, e.code() = 2, e.displayText() = Exception: EstablishDisaggTask Failed13: , e.what() = Exception
```

GRPC Code 13 suggests an internal error happens.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Disaggregated/RNPageReceiverContext.cpp [hunk 1]
```diff
@@ -99,9 +102,36 @@ FetchPagesRequest::FetchPagesRequest(DM::RNRemoteSegmentReadTaskPtr seg_task_)
...
+            auto page_oid = DM::Remote::PageOID{
+                .store_id = seg_task->store_id,
+                .table_id = seg_task->table_id,
+                .page_id = page_id,
+            };
+            persisted_oids.emplace_back(page_oid);
+        }
+
+        auto page_cache = seg_task->dm_context->db_context.getSharedContextDisagg()->rn_page_cache;
+        auto occupy_result = page_cache->occupySpace(persisted_oids, seg_task->delta_tinycf_page_sizes);
```
Comment (JaySon-Huang) [inline dbms/src/Flash/Disaggregated/RNPageReceiverContext.cpp:120]:
```suggestion
        std::vector<DM::Remote::PageOID> to_read_oids;
        to_read_oids.reserve(seg_task->delta_tinycf_page_ids.size());
        for (const auto & page_id : seg_task->delta_tinycf_page_ids)
        {
            auto page_oid = DM::Remote::PageOID{
                .store_id = seg_task->store_id,
                .table_id = seg_task->table_id,
                .page_id = page_id,
            };
            to_read_oids.emplace_back(page_oid);
        }

        auto page_cache = seg_task->dm_context->db_context.getSharedContextDisagg()->rn_page_cache;
        auto occupy_result = page_cache->occupySpace(to_read_oids, seg_task->delta_tinycf_page_sizes);
```
name it as `to_read_oids`? `persisted_oids` make me think they have already been persisted in RN

Comment (JaySon-Huang) [inline dbms/src/Flash/Disaggregated/RNPageReceiverContext.cpp:120]:
```suggestion
        // occupy space segment by segment
        auto page_cache = seg_task->dm_context->db_context.getSharedContextDisagg()->rn_page_cache;
        auto occupy_result = page_cache->occupySpace(persisted_oids, seg_task->delta_tinycf_page_sizes);
```

**Review by JaySon-Huang — APPROVED** [review]
> Close https://github.com/pingcap/tiflash/issues/7052 (but I have no idea why it is fixed)

I also do not understand why it only appears under disagg mode? But let' make it work now

LGTM

File: dbms/src/Flash/Disaggregated/RNPageReceiverContext.cpp (hunk 1 above)
Comment (breezewish) [reply dbms/src/Flash/Disaggregated/RNPageReceiverContext.cpp:120]: Yes, it is a naming mistake. Thanks for pointing out!

**Pull request merged**